File: GCDUserInterface/ChangeDetection/DoDSummaryDisplayOptions.vb
Namespace ChangeDetection

    Public Class DoDSummaryDisplayOptions

        Public Enum RowGroups
            ShowAll
            Normalized
            SpecificGroups
        End Enum

        Private m_Units As GCDConsoleLib.GCD.UnitGroup
        Public m_nPrecision As Integer

        Public m_eRowGroups As RowGroups
        Public m_bRowsAreal As Boolean
        Public m_bRowsVolumetric As Boolean
        Public m_bRowsVerticalAverages As Boolean
        Public m_bRowsPercentages As Boolean

        Public m_bColsRaw As Boolean
        Public m_bColsThresholded As Boolean
        Public m_bColsPMError As Boolean
        Public m_bColsPCError As Boolean

        Public ReadOnly Property Units As GCDConsoleLib.GCD.UnitGroup
            Get
                Return m_Units
            End Get
        End Property

        Public Property LinearUnits As UnitsNet.Units.LengthUnit
            Get
                Return m_Units.VertUnit
            End Get
            Set(value As UnitsNet.Units.LengthUnit)
                m_Units = New GCDConsoleLib.GCD.UnitGroup(m_Units.VolUnit, m_Units.ArUnit, value, m_Units.HorizUnit)
            End Set
        End Property

        Public Property AreaUnits As UnitsNet.Units.AreaUnit
            Get
                Return m_Units.ArUnit
            End Get
            Set(value As UnitsNet.Units.AreaUnit)
                m_Units = New GCDConsoleLib.GCD.UnitGroup(m_Units.VolUnit, value, m_Units.VertUnit, m_Units.HorizUnit)
            End Set

        End Property

        Public Property VolumeUnits As UnitsNet.Units.VolumeUnit
            Get
                Return m_Units.VolUnit
            End Get
            Set(value As UnitsNet.Units.VolumeUnit)
                m_Units = New GCDConsoleLib.GCD.UnitGroup(value, m_Units.ArUnit, m_Units.VertUnit, m_Units.HorizUnit)
            End Set
        End Property

        Public Sub New(dataUnits As GCDConsoleLib.GCD.UnitGroup)

            m_Units = dataUnits
            m_nPrecision = 2

            m_eRowGroups = RowGroups.ShowAll
            m_bRowsAreal = True
            m_bRowsVolumetric = True
            m_bRowsVerticalAverages = True
            m_bRowsPercentages = True

            m_bColsRaw = True
            m_bColsThresholded = True
            m_bColsPMError = True
            m_bColsPCError = True

        End Sub

    End Class

End Namespace

File: GCDCore/UserInterface/ChangeDetection/DoDSummaryDisplayOptions.cs
namespace GCDCore.UserInterface.ChangeDetection
{
    public class DoDSummaryDisplayOptions
    {
        public enum RowGroups
        {
            ShowAll,
            Normalized,
            SpecificGroups
        }

        private GCDConsoleLib.GCD.UnitGroup m_Units;

        public int m_nPrecision;
        public RowGroups m_eRowGroups;
        public bool m_bRowsAreal;
        public bool m_bRowsVolumetric;
        public bool m_bRowsVerticalAverages;

        public bool m_bRowsPercentages;
        public bool m_bColsRaw;
        public bool m_bColsThresholded;
        public bool m_bColsPMError;
        public bool m_bColsPCError;

        public System.Drawing.Font Font { get; set; }

        public bool AutomatedYAxisScale { get; set; }

        public GCDConsoleLib.GCD.UnitGroup Units { get { return m_Units; } }

        public UnitsNet.Units.LengthUnit LinearUnits
        {
            get { return m_Units.VertUnit; }
            set { m_Units = new GCDConsoleLib.GCD.UnitGroup(m_Units.VolUnit, m_Units.ArUnit, value, m_Units.HorizUnit); }
        }

        public UnitsNet.Units.AreaUnit AreaUnits
        {
            get { return m_Units.ArUnit; }
            set { m_Units = new GCDConsoleLib.GCD.UnitGroup(m_Units.VolUnit, value, m_Units.VertUnit, m_Units.HorizUnit); }
        }


        public UnitsNet.Units.VolumeUnit VolumeUnits
        {
            get { return m_Units.VolUnit; }
            set { m_Units = new GCDConsoleLib.GCD.UnitGroup(value, m_Units.ArUnit, m_Units.VertUnit, m_Units.HorizUnit); }
        }

        public DoDSummaryDisplayOptions(GCDConsoleLib.GCD.UnitGroup dataUnits)
        {
            m_Units = dataUnits;
            m_nPrecision = 2;

            m_eRowGroups = RowGroups.ShowAll;
            m_bRowsAreal = true;
            m_bRowsVolumetric = true;
            m_bRowsVerticalAverages = true;
            m_bRowsPercentages = true;

            m_bColsRaw = true;
            m_bColsThresholded = true;
            m_bColsPMError = true;
            m_bColsPCError = true;

            Font = Properties.Settings.Default.ChartFont;

            AutomatedYAxisScale = true;
        }
    }
}